File: Python/Program/Cpu.py
from Komponen import Komponen

class Cpu(Komponen):
    # Constructor
    def __init__(self, merk="", nama="", jumlah_core=0, kecepatan_GHz=0.0):
        super().__init__(merk, nama)
        self.jumlah_core = jumlah_core
        self.kecepatan_GHz = kecepatan_GHz
    
    # Getters
    def get_jumlah_core(self):
        return self.jumlah_core
    
    def get_kecepatan_GHz(self):
        return self.kecepatan_GHz
    
    # Setters
    def set_jumlah_core(self, jumlah_core):
        self.jumlah_core = jumlah_core
    
    def set_kecepatan_GHz(self, kecepatan_GHz):
        self.kecepatan_GHz = kecepatan_GHz

File: Java/Program/Cpu.java
public class Cpu extends Komponen {
    // Attributes
    private int jumlahCore;
    private float kecepatanGHz;
    
    // Constructors
    public Cpu() {
        this.jumlahCore = 0;
        this.kecepatanGHz = 0.0f;
    }
    
    public Cpu(String merk, String nama, int jumlahCore, float kecepatanGHz) {
        super(merk, nama);
        this.jumlahCore = jumlahCore;
        this.kecepatanGHz = kecepatanGHz;
    }
    
    // Getters
    public int getJumlahCore() {
        return this.jumlahCore;
    }
    
    public float getKecepatanGHz() {
        return this.kecepatanGHz;
    }
    
    // Setters
    public void setJumlahCore(int jumlahCore) {
        this.jumlahCore = jumlahCore;
    }
    
    public void setKecepatanGHz(float kecepatanGHz) {
        this.kecepatanGHz = kecepatanGHz;
    }
}
